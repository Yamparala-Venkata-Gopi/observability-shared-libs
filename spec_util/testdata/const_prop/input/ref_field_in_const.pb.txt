# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    ref: {
      method_index: 0
      arg_ref: {
        # References a field in a constant struct.
        key: "struct_constant_arg"
        data_ref: {
          struct_ref: {
            field_ref: {
              key: "foo"
              data_ref: {
                primitive_ref: {}
              }
            }
          }
        }
      }
    }
  }
}

arg_templates: {
  key: "arg2"
  value: {
    ref: {
      method_index: 0
      arg_ref: {
        # References an elem in a constant list.
        key: "list_constant_arg"
        data_ref: {
          list_ref: {
            elem_ref: {
              index: 1
              data_ref: {
                primitive_ref: {}
              }
            }
          }
        }
      }
    }
  }
}
