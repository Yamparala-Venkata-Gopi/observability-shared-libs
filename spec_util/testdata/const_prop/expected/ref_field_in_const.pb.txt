# MethodTemplate proto

method_id: {
  name: "new_method_with_refs"
}

arg_templates: {
  key: "arg1"
  value: {
    value: {
      primitive: {
        type_hint: "string_constant_2"
        string_value: {
          value: "1aa4e597-a29d-42d4-8134-3e4ee73aa7a1"
        }
      }
    }
  }
}

arg_templates: {
  key: "arg2"
  value: {
    value: {
      primitive: {
        type_hint: "string_constant_5"
        string_value: {
          value: "8836f293-448a-4db8-9ed4-b19e8211f9b8"
        }
      }
    }
  }
}
